### PR TITLE
rustdoc: update `pulldown-cmark` to 0.10

### PR DESCRIPTION
**Pull request by stepantubanov**
Updated from 0.9 to 0.10.

### CODE REVIEW

File: src/librustdoc/html/markdown.rs [hunk 1]
```diff
@@ -442,7 +435,8 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for LinkReplacer<'a, I> {
             }
             // If this is a link, but not a shortcut link,
             // replace the URL, since the broken_link_callback was not called.
-            Some(Event::Start(Tag::Link(_, dest, title))) => {
+            Some(Event::Start(Tag::Link { dest_url: dest, title, .. })) => {
+                assert!(self.shortcut_link.is_none(), "links cannot be nested");
```
Comment (stepantubanov) [inline src/librustdoc/html/markdown.rs:439]: Not completely sure about this one. Added it to make sure we're not silently mismatching link start/end link tags.

**Comment by GuillaumeGomez** [comment]
cc @notriddle

**Comment by stepantubanov** [comment]
@notriddle I see, that makes sense, thanks! Should I close this or keep it open until we can update?

**Comment by Dylan-DPC** [comment]
@stepantubanov if you can resolve the conflict with the lock file and mark this as ready for review we can push this further

**Pull request closed without merging**